src/query_backend.h

```diff
@@ -1082,11 +1004,26 @@ class Sha1 {
     }
   }
 
-  /// \}
 
   Digest digest() {
     Digest d;
-    hash_.get_digest(d.val);
+    #if BOOST_VERSION_MINOR < 86
+        unsigned int tmp[5];
+    #else
+        unsigned char tmp[20];
+    #endif
```

Comment:
I'm not sure I understand all of what's happening behind the scenes here, but it seems that it might always work to use the `unsigned char tmp[20]` option and unpack it into the `Digest` as you do below??

Reply:
On Boost<1.86 `boost::uuids::detail::sha1::get_digest(tmp)` is declared with tmp as `unsigned int[5]`, but on Boost==1.86 `get_digest(tmp)` is declared with tmp as `unsigned char[20]`, so if we try to always declare `tmp` as a `unsigned char[20]` this will fail to compile on Boost<1.86.  

Prior to Boost 1.86, we passed the digest around as is, so our hdf5 backend infrastructure is built to handle `uint[5]` (its actually pretty generic, so long as the dimensions arent [too high](https://github.com/cyclus/cyclus/pull/1792#issuecomment-2458665109), but lets leave it as `uint[5]` since that seems to work well).  On Boost==1.86 we can no longer pass it around as is without modifying our hdf5 infrastructure, so this converts the `unsigned char[20]` array  to `unsigned int[5]` to make it compatible.  In theory we could use an arbitrary type/dimension `atype[dim]` for our hdf5 backend infrastructur, so long as we convert it to `atype[dim]` *from* the format that `get_digest()` requires (either `uint[5]` or `uchar[20]`).


Does this make sense or am I overcomplicating things?

Reply:
That's useful.  I had was stuck thinking we could just pass in a pointer to some allocated memory, and didn't think of the Boost signature being so strongly typed.